tls/s2n_shutdown.c

```diff
@@ -48,8 +48,10 @@ int s2n_shutdown(struct s2n_connection *conn, s2n_blocked_status * more)
         conn->in_status = ENCRYPTED;
     }
 
-    /* Fails with S2N_ERR_SHUTDOWN_RECORD_TYPE or S2N_ERR_ALERT on receipt of anything but a close_notify */
-    POSIX_GUARD(s2n_recv_close_notify(conn, more));
+    if (!conn->closed) {
+      /* Fails with S2N_ERR_SHUTDOWN_RECORD_TYPE or S2N_ERR_ALERT on receipt of anything but a close_notify */
+      POSIX_GUARD(s2n_recv_close_notify(conn, more));
+    }
```

Comment:
Is it possible to add tests to confirm this behavior and error codes?

Reply:
the current tests fail so i am trying to understand the failures first. also ill mark this pr as a WIP